### DISCUSSION TITLE
About Mesh

### DISCUSSION THREAD

**Discussion by wowodejiajia** [idaholab/moose#25491] [Q&A General] [answered]
Dear MOOSE experts,
```
[Mesh]
  type = GeneratedMesh
  dim = 2
  nx = 14
  ny = 14
  xmax = 9
  ymax = 9
  uniform_refine = 3
[]
```
I don't understand what "uniform_refine = 3" means.  What is the effect on mesh establishment? What is the difference between uniform_refine = 3 and uniform_refine = 2?

Thank you.

**Comment by mangerij** [comment, marked as answer]
If you have nx = 14, ny = 14 means you have 14 elements along the x direction and 14 elements along the y direction.

Uniform refine = 1 will make nx = 28, ny = 28
Uniform refine = 2 will make nx = 56, ny = 56
Uniform refine = 3 will make nx = 128, ny = 128

This happens at the mesh setup step.

This changes in 3D when the refinement also has another dimension to do

**Reply by wowodejiajia** [reply]
I understand. Thank you very much

**Reply by ykvishal** [reply]
I think that it should be  `Uniform refine = 3 will make nx = 112, ny = 112` instead of  `Uniform refine = 3 will make nx = 128, ny = 128` , The `uniform refine = n` option changes mesh by `nx * 2^n`
